### PR TITLE
:art: :lipstick: Created a reusable pricing card component #81

### LINKED ISSUE #81: 🎨 💄 Create A Reusable Pricing Card Component
#  A Reusable Pricing Card Component
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a developer, I want a reusable pricing card component I can use to render an individual price card.
I want a good experience on the web and phone.

### Description

Design:

pricing one

pricing two

[Link to full design](https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528)

### Technical Analysis

The reusable price item component should be placed in the library `libs/elements/cards` (cards is a module that already exists)

It exports the elewa-website-price-item-card component, for other modules to import. The component name is elewa-website-price-item-card.

The component accepts an Angular input of type:

```typeScript
interface PriceItem extends PriceItemColors {
  title: string
  amount: number
  subTitle: string
  description: string
  sliderButton: SliderButtonData
}

interface PriceItemColors {
  amountColor: string
  textColor: string
  backgroundColor: string
}
```

The `SliderButtonData` type already exists in the codebase at `libs/models/schema/ui/buttons`.

The type/model should be stored in `libs/models/schema/ui/cards` (cards is a module/library already implemented in the codebase). The name of the file containing the interface is price-item.interface.ts and it's exported by its parent module above.

### Acceptance Criteria

- [ ] When rendered the component is pixel-perfect compared to the design above.
- [ ] The component is clean and does not have more than 50 lines
- [ ] The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
- [ ] The code is isolated to the `libs/elements/layout/cards` package unless a justified reason is specified.
- [ ] The model/schema is isolated in the `libs/models/schema/ui/cards` package unless a justified reason is specified.
- [ ] The design is responsive for screen, tablet and phone
- [ ] A separate commit is used for each small change
- [ ] The commits and PR are well formatted/use proper documentation
- [ ] The module exports the component

### PR DESCRIPTION
**Pull request by levy-web**
# Description

Build a re usable card component that uses slider button data type which and also priceItem type where thr slider button data is imported and used among the variables

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes #81

## Type of change

Please delete options that are not relevant.

- [X ] New feature (non-breaking change which adds functionality)

## Screenshot (optional)
**re - used 3 times in display flex**

Screenshot from 2023-08-23 09-31-01

**On Mobile**
Screenshot from 2023-08-23 09-33-24
Screenshot from 2023-08-23 09-33-02

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/elements/cards/src/lib/cards.module.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { ElewaInfoCardComponent } from './elewa-info-card/elewa-info-card.component';
+import { ElewaWebsitePriceItemCardComponent } from './elewa-website-price-item-card/elewa-website-price-item-card.component';
+import { ButtonsModule } from '@elewa-website/elements/layout/buttons';
```
Comment (LemmyMwaura) [inline libs/elements/cards/src/lib/cards.module.ts:5]: Fix the order of imports. Domain imports come before local imports.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Very clean work 👍. Just one fix and one challenge 🧱. The challenge is optional

File: libs/elements/cards/src/lib/elewa-website-price-item-card/elewa-website-price-item-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    return number.toLocaleString('en-US');
+  }
+
+  // Property to store the formatted price string
+  formattedPrice = '';
+
+  ngOnChanges() {
+    if (this.priceItem) {
+      this.formattedPrice = this.formatPriceWithCommas(this.priceItem.amount);
+    }
```
[the remark this answers is not in the document]
Comment (levy-web) [reply libs/elements/cards/src/lib/elewa-website-price-item-card/elewa-website-price-item-card.component.ts]: @LemmyMwaura I have attempted the challenge you can review.

**Review by LemmyMwaura — APPROVED** [review]

File: libs/elements/cards/src/lib/elewa-website-price-item-card/elewa-website-price-item-card.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (LemmyMwaura) [reply libs/elements/cards/src/lib/elewa-website-price-item-card/elewa-website-price-item-card.component.ts]: It's good ✅

**Pull request merged**